### ISSUE TITLE
Verify PSC DNS name is present as a SAN in the server certificate

### ISSUE THREAD

**Issue by nancynh** [GoogleCloudPlatform/alloydb-python-connector#324]
Verify PSC DNS name is present as a SAN in the server certificate. It means removing the existing workaround to check the private IP address instead of the DNS name (https://github.com/GoogleCloudPlatform/alloydb-python-connector/pull/291).

**Comment by jackwotherspoon** [comment]
is this a duplicate of #313 ?

**Comment by enocom** [comment]
Yes, let's use this one for tracking.

**Comment by jackwotherspoon** [comment]
Good context on the trailing dot being stripped https://github.com/dpkp/kafka-python/pull/2374#issuecomment-1669886921

**Comment by jackwotherspoon** [comment]
More good context on this thread: https://github.com/openssl/openssl/issues/11560

**Comment by enocom** [comment]
Thanks to @jackwotherspoon for finding this:

https://github.com/openssl/openssl/issues/11560#issuecomment-1631520748

TL;DR: SANs shouldn't have trailing dots.

**Comment by enocom** [comment]
RFC 6066 says [no trailing dots in SANs](https://datatracker.ietf.org/doc/html/rfc6066#:~:text=The%20hostname%20is%20represented%20as%20a%20byte%0A%20%20%20string%20using%20ASCII%20encoding%20without%20a%20trailing%20dot).

**Comment by rhatgadkar-goog** [comment]
Fixed in https://github.com/GoogleCloudPlatform/alloydb-python-connector/pull/446
